diff --git a/.gitignore b/.gitignore
index 0f23d8a..62b0739 100644
--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,7 @@
 
 # direnv cache
 .direnv
+
+.idea
+
+Cargo.lock
diff --git a/Cargo.lock b/Cargo.lock
index 55fcde9..5269b40 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3806,6 +3806,7 @@ dependencies = [
  "pallet-aura",
  "pallet-balances",
  "pallet-grandpa",
+ "pallet-poe",
  "pallet-randomness-collective-flip",
  "pallet-sudo",
  "pallet-template",
@@ -4042,6 +4043,21 @@ dependencies = [
  "sp-std",
 ]
 
+[[package]]
+name = "pallet-poe"
+version = "4.0.0-dev"
+dependencies = [
+ "frame-benchmarking",
+ "frame-support",
+ "frame-system",
+ "parity-scale-codec",
+ "scale-info",
+ "sp-core",
+ "sp-io",
+ "sp-runtime",
+ "sp-std",
+]
+
 [[package]]
 name = "pallet-randomness-collective-flip"
 version = "4.0.0-dev"
diff --git a/Cargo.toml b/Cargo.toml
index 538fd8d..a1804b4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,6 +2,7 @@
 members = [
     "node",
     "pallets/template",
+	"pallets/poe",
     "runtime",
 ]
 [profile.release]
diff --git a/runtime/Cargo.toml b/runtime/Cargo.toml
index c8d0b4d..32a509a 100644
--- a/runtime/Cargo.toml
+++ b/runtime/Cargo.toml
@@ -49,6 +49,7 @@ frame-system-benchmarking = { version = "4.0.0-dev", default-features = false, g
 
 # Local Dependencies
 pallet-template = { version = "4.0.0-dev", default-features = false, path = "../pallets/template" }
+pallet-poe = { version = "4.0.0-dev", default-features = false, path = "../pallets/poe" }
 
 [build-dependencies]
 substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30" }
@@ -72,6 +73,7 @@ std = [
 	"pallet-randomness-collective-flip/std",
 	"pallet-sudo/std",
 	"pallet-template/std",
+	"pallet-poe/std",
 	"pallet-timestamp/std",
 	"pallet-transaction-payment-rpc-runtime-api/std",
 	"pallet-transaction-payment/std",
diff --git a/runtime/src/lib.rs b/runtime/src/lib.rs
index f801068..ee4d4f2 100644
--- a/runtime/src/lib.rs
+++ b/runtime/src/lib.rs
@@ -275,6 +275,11 @@ impl pallet_template::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 }
 
+impl pallet_poe::Config for Runtime {
+	type MaxClaimLength = ConstU32<512>;
+	type RuntimeEvent = RuntimeEvent;
+}
+
 // Create the runtime by composing the FRAME pallets that were previously configured.
 construct_runtime!(
 	pub struct Runtime
@@ -293,6 +298,7 @@ construct_runtime!(
 		Sudo: pallet_sudo,
 		// Include the custom logic from the pallet-template in the runtime.
 		TemplateModule: pallet_template,
+		PoeModule: pallet_poe,
 	}
 );
 
